### DISCUSSION TITLE
Purchase Order Lines

### DISCUSSION THREAD

**Discussion by lokyi792** [inventree/InvenTree#7654] [Q&A] [answered]
Purchase Order lines make absolutely no sense to me with quantity pack sizes <1 and the sample P/O code.
Firstly the column says quantity. Ok, Is that individual units I want to buy or number of packs? (I guessed packs) and when I hover over it, it says Pack Quantity = X Total Quantity = Y. Ok. 
Now the price column says explicitly "Unit Price". Which I assume to be the price of each unit. But from the calculation shown it isn't, it's the package price? If so, does this save back the per unit price or am I completely on the wrong path here?

Also in the new UI there is literally no way I can see to print the P/O?

![2024-07-15_15-17-34](https://github.com/user-attachments/assets/3b535d6c-b46d-4123-a1c7-dc7129787309)

**Comment by SchrodingersGat** [comment]
@lokyi792 the unit price here is the price of each "pack". When the item is received, that price is converted back into a unit price for the received stock, divided through by the supplier part pack quantity. 

I'll look into printing purchase order reports in the new interface.

**Reply by lokyi792** [reply]
Awesome, is it possible to pull the supplier part packaging field through to put in the table after the quantity field on the purchase order?
![2024-07-15_16-17-24](https://github.com/user-attachments/assets/8c44b81f-3963-4a22-8393-a58413f5025d)

**Reply by SchrodingersGat** [reply]
@lokyi792 implementing this here - https://github.com/inventree/InvenTree/pull/7656

**Comment by SchrodingersGat** [comment]
Regarding printing purchase order reports in the new interface:

![image](https://github.com/user-attachments/assets/a820adff-2d46-434e-b8c9-8d8b864a69cf)

**Reply by lokyi792** [reply]
Ain't no sunshine there... also the Dashboard/Parts/Stock tabs don't follow the user permissions either, they all show regardless. Work perfectly in old UI though.
![2024-07-19_15-50-56](https://github.com/user-attachments/assets/7982c2de-cd35-4bba-be97-48684146acc2)

**Reply by SchrodingersGat** [reply]
@lokyi792 what version of InvenTree are you running? Many (most) of the new UI features are only available in the master branch, are you running the stable release?

**Reply by lokyi792** [reply]
15.5?

**Reply by SchrodingersGat** [reply]
@lokyi792 ok, then please compare to the functionality on the demo server which is running the "latest" code branch (pending the `0.16.0` release:

https://demo.inventree.org/platform/

**Reply by matmair** [reply]
A lot of new UI things are not included there. Can you check on the demo? That is pretty recent.

**Reply by lokyi792** [reply, marked as answer]
So demo server is 0.16 dev? That all looks fine to me. Printing icon is there.
